lib/ocpp/v201/charge_point.cpp

```diff
@@ -1997,13 +1998,20 @@ AuthorizeResponse ChargePoint::authorize_req(const IdToken id_token, const std::
     req.certificate = certificate;
     req.iso15118CertificateHashData = ocsp_request_data;
 
+    AuthorizeResponse response;
```

Comment:
We should never come here if we are not connected. That still does not mean we might not run into timeout issues in the cases you mentioned so I do think it is valid to have a timeout here.

We should add an additional check for connected here I would say.